tests/navigation.el

```diff
@@ -44,6 +44,14 @@
 (ert-deftest js2-jump-to-object-property ()
   (js2-navigation-helper "var aObject = {prop1: 3, prop2: \"hello\"}; aObject.prop1" 16))
 
+(ert-deftest js2-jump-to-object-property-with-competitor-object ()
+  (js2-navigation-helper "var foo = {bar: 42}; var bar = {bar: 24}; foo.bar" 12))
+
+(ert-deftest js2-jump-to-object-property-with-competitor-function-constructor ()
+  "Currently this a known misbehaviour."
+  :expected-result :failed
+  (js2-navigation-helper "var foo = {bar: 42}; function Foo() { this.bar = 0; }; foo.bar" 12))
```

Comment:
Why does this happen? Is it because we're not tracking the match kinds (exact vs. "any object can be `this`")?

I'd expect the first match to win out anyway though, right?